akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala

```diff
@@ -77,6 +78,32 @@ object Sink extends SinkApply {
   def head[T]: Sink[T, Future[T]] = new Sink(new HeadSink[T](DefaultAttributes.headSink, shape("HeadSink")))
 
   /**
+   * A `Sink` that materializes into a `Future` of all elements in the stream
+   * as a strict collection.
+   * If the stream is inifite, this will never complete but it will consume
+   * all of your memory.
+   */
+  def toSeq[T](implicit ec: ExecutionContext): Sink[T, Future[immutable.Seq[T]]] =
```

Comment:
this method should not take an ExecutionContext: the recovery step is wait-free and therefore can run on the sameThreadExecutionContext